### PR TITLE
AutoShell Refactoring

### PR DESCRIPTION
**Pull request by TalZaccai**
Structural refactoring of `autoShell` to make the codebase testable, maintainable, and extensible. All 78 commands continue to work identically — verified by a command-by-command audit against `origin/main`.

---

### Phase 1: Core Architecture

- [x] Extract 10 OS abstraction interfaces (`IAudioService`, `IRegistryService`, `ISystemParametersService`, `IProcessService`, `IWindowService`, `INetworkService`, `IBrightnessService`, `IDisplayService`, `IVirtualDesktopService`, `IDebuggerService`)
- [x] Extract 17 command handler classes by domain (`AudioCommandHandler`, `WindowCommandHandler`, `NetworkCommandHandler`, 9 settings sub-handlers, etc.)
- [x] Replace 78-case `execLine` switch with `CommandDispatcher` + dictionary lookup
- [x] Extract shared app data into `IAppRegistry` service (replaces static fields)
- [x] Migrate all scenario-specific logic from `AutoShell.cs` into handlers (~1604 → ~80 lines)
- [x] Move all P/Invoke declarations into their respective domain handlers/services
- [x] Move all COM interop (Virtual Desktop, Audio, Network) into respective services
- [x] Move interop files to `Services/Interop/` (`CoreAudioInterop.cs`, `UIAutomation.cs`)
- [x] Move `IAppRegistry` + `WindowsAppRegistry` to project root
- [x] Delete `AutoShell_Win32.cs`, `AutoShell_Settings.cs`, `AutoShell_Themes.cs` — all logic now in handlers/services
- [x] Delete legacy `packages.config` and `App.config`
- [x] Add centralized error handling (try/catch in dispatcher)
- [x] Handle `quit` directly in dispatcher (not in handlers)

### Phase 2: Code Quality

- [x] Add XML doc comments to all interface methods + `<inheritdoc/>` on implementations
- [x] Clean up unused using directives across all files
- [x] Add explicit private modifiers (satisfying `.editorconfig` rules)
- [x] Unify all command names to PascalCase (C# handlers + TS schemas + connector)
- [x] Run `dotnet format` for code style consistency
- [x] Fix all build warnings (CA1806, SYSLIB1054) — target zero warnings
- [x] Modernize collections: `Hashtable` → `Dictionary<string,string>` with `OrdinalIgnoreCase`, `SortedList` → `Dictionary`
- [x] Fix `Dictionary` indexer regression (TryGetValue for null-return contract)
- [x] Add `ILogger` abstraction with `Error`, `Warning`, `Info`, `Debug` levels
- [x] Replace `Console.WriteLine`/`Debug.WriteLine` with proper logger calls
- [x] Restore all important original comments (API semantics, design rationale, TODOs)

### Phase 3: Testing

- [x] Create `autoShell.Tests` xUnit project with Moq
- [x] Unit tests for all 17 handler classes with mocked services (153 tests)
- [x] `CommandDispatcher` tests — verify correct handler routing for each action
- [x] `HandlerRegistration` reflection tests — enforce every command has at least one unit test
- [x] `WindowsAppRegistry` tests — real class, null returns, case insensitivity (9 tests)
- [x] `CommandDispatcherIntegration` tests — full wiring with mock services (11 tests)
- [x] E2E stdin/stdout tests — query commands, protocol edge cases, command-line mode (15 tests)
- [x] Add `dotnet test` to CI

### CODE REVIEW

File: dotnet/autoShell/Services/WindowsNetworkService.cs [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+    [DllImport("wlanapi.dll")]
+    private static extern void WlanFreeMemory(IntPtr pMemory);
+
+    [DllImport("wlanapi.dll")]
+    private static extern int WlanConnect(IntPtr hClientHandle, ref Guid pInterfaceGuid, ref WLAN_CONNECTION_PARAMETERS pConnectionParameters, IntPtr pReserved);
+
+    [DllImport("wlanapi.dll")]
+    private static extern int WlanDisconnect(IntPtr hClientHandle, ref Guid pInterfaceGuid, IntPtr pReserved);
+
+    [DllImport("wlanapi.dll")]
```
Comment (robgruen) [inline dotnet/autoShell/Services/WindowsNetworkService.cs]: Nit, you could use constants for the library name:


    private const string WLAN_API = "wlanapi.dll";

    [DllImport(WLAN_API, SetLastError = true)]

**Review by robgruen — APPROVED** [review]
Love this PR, this is SO much nicer for many reasons.  Thank you for doing this!
